llrt_core/src/modules/http/response.rs

```diff
@@ -149,7 +166,21 @@ impl<'js> Response<'js> {
                     body.body_mut().collect().await.or_throw(ctx)?.to_bytes()
                 };
 
-                bytes.to_vec()
+                if self.body_attributes.content_encoding.is_none() {
```

Comment:
We can do the other way around and do `if let Some(encoding) =  self.body_attributes.content_encoding {...}` which would avoid the unwrap below.

Reply:
Thank you for pointing this out! 
It would be a more prospective code. I will present it in the next commit.